cromwell-drs-localizer/src/main/scala/drs/localizer/downloaders/Downloader.scala

```diff
@@ -2,8 +2,16 @@ package drs.localizer.downloaders
 
 import cats.effect.{ExitCode, IO}
 
-trait Downloader {
-
-  def download: IO[ExitCode]
+sealed trait DownloadResult {
+  def exitCode: ExitCode
+}
+case object DownloadSuccess extends DownloadResult {
+  override def exitCode: ExitCode = ExitCode(0)
+}
+sealed trait DownloadFailure extends DownloadResult
+case class RetryableDownloadFailure(override val exitCode: ExitCode = ExitCode(0)) extends DownloadFailure
+case class NonRetryableDownloadFailure(override val exitCode: ExitCode = ExitCode(0)) extends DownloadFailure
```

Comment:
Curiosity: why have defaults here? Do failures with `ExitCode(0)` have an implicit meaning?

Reply:
I think this is leftover from an earlier version of the design, certainly not needed now.